radix-transactions/src/manifest/static_resource_movements/effect.rs

```diff
@@ -0,0 +1,1541 @@
+                            AccountMethod::Deposit(input) => input.output(node_id, inputs, instruction_index),
+                            AccountMethod::Withdraw(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::WithdrawNonFungibles(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::LockFeeAndWithdraw(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::LockFeeAndWithdrawNonFungibles(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::CreateProofOfAmount(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::CreateProofOfNonFungibles(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::SetDefaultDepositRule(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::SetResourcePreference(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::RemoveResourcePreference(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::TryDepositOrAbort(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::Burn(input) => input.output(node_id, inputs, instruction_index),
+                            AccountMethod::BurnNonFungibles(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::AddAuthorizedDepositor(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::RemoveAuthorizedDepositor(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            AccountMethod::TryDepositOrRefund(input) => {
+                                input.output(node_id, inputs, instruction_index)
+                            }
+                            // TODO: A bit of a hack to support expressions. We need a better way to
```

Comment:
(Removed in https://github.com/radixdlt/radixdlt-scrypto/pull/1931)

Reply:
It's because we can't SBOR decode expressions as `Account*DepositBatch*ManifestInput`.

Here's an example. Imagine that we have the instruction `CALL_METHOD Address("account_rdx1...") "deposit_batch" Expression("ENTIRE_WORKTOP");`. The `AccountDepositBatchManifestInput` is defined as the following:

```rust
#[derive(Debug, Eq, PartialEq, ManifestSbor)]
pub struct AccountDepositBatchManifestInput {
    pub buckets: Vec<ManifestBucket>,
}
```

We can't SBOR decode the `Expression("ENTIRE_WORKTOP")` as `Vec<ManifestBucket>`. Therefore, the SBOR decoding fails.